appeals/web/src/server/appeals/appeal-details/interested-party-comments/interested-party-comments.mapper.js

```diff
@@ -207,63 +207,74 @@ function generateCommentSummaryList(comment) {
 			  )
 			: null;
 
+	const rows = [
+		{
+			key: { text: 'Interested party' },
+			value: { text: comment.represented.name }
+		},
+		{
+			key: { text: 'Email' },
+			value: { text: comment.represented.email }
+		},
+		{
+			key: { text: 'Address' },
+			value: hasAddress
+				? { html: addressToMultilineStringHtml(comment.represented.address) }
+				: { text: 'Not provided' },
+			actions: {
+				items: [
+					hasAddress
+						? { text: 'Change', href: '#', visuallyHiddenText: 'address' }
+						: { text: 'Add', href: '#', visuallyHiddenText: 'address' }
+				]
+			}
+		},
+		{
+			key: { text: 'Submitted' },
+			value: { html: dateISOStringToDisplayDate(comment.created) }
+		},
+		{
+			key: { text: comment.redactedRepresentation ? 'Original comment' : 'Comment' },
+			value: {
+				text: commentIsDocument ? 'Added as a document' : comment.originalRepresentation
+			},
+			actions: {
+				items: commentIsDocument ? [] : [{ text: 'Redact', href: '#' }]
+			}
+		},
+		...(comment.redactedRepresentation
+			? [{ key: { text: 'Redacted comment' }, value: { text: comment.redactedRepresentation } }]
+			: []),
+		{
+			key: { text: 'Supporting documents' },
+			value: attachmentsList ? { html: attachmentsList } : { text: 'Not provided' },
+			actions: {
+				items: [
+					...(comment.attachments?.length > 0
+						? [{ text: 'Manage', href: '#', visuallyHiddenText: 'supporting documents' }]
+						: []),
+					{ text: 'Add', href: '#', visuallyHiddenText: 'supporting documents' }
+				]
+			}
+		}
+	];
+
+	if (!isReviewPage) {
```

Comment:
There's nothing wrong with this approach, but in a few other places where we've needed to do something similar, we've used the same conditional destructuring approach as with the `comment.redactedRepresentation` part a little further up this function body, rather than using `splice`. There's another example here for reference: https://github.com/Planning-Inspectorate/appeals-back-office/blob/main/appeals/web/src/server/lib/mappers/appellant-case/appeal-type-has.mapper.js#L61